cupy/core/include/cupy/tuple/tuple.h

```diff
@@ -0,0 +1,961 @@
+/*
```

Comment:
Maybe using jitify instead of forking the header works?

Reply:
However, I am not sure if jitify worked with amd

Reply:
No, it doesn't work, we checked together.

Reply:
Well, another route for AMD is to compile using hipcc instead of hiprtc if tuple is needed. I didn't test it though.